win32ss/user/user32/windows/input.c

```diff
@@ -143,6 +143,107 @@ CliImmSetHotKeyWorker(DWORD dwHotKeyId, UINT uModifiers, UINT uVirtualKey, HKL h
     return FALSE;
 }
 
+BOOL APIENTRY
+CliSaveImeHotKey(DWORD dwID, UINT uModifiers, UINT uVirtualKey, HKL hKL, BOOL bDelete)
+{
+    WCHAR szName[MAX_PATH];
+    LONG error;
+    HKEY hControlPanel = NULL, hInputMethod = NULL, hHotKeys = NULL, hKey = NULL;
+    BOOL ret = FALSE, bRevertOnFailure = FALSE;
+
+    if (bDelete)
+    {
+        StringCchPrintfW(szName, _countof(szName),
+                         L"Control Panel\\Input Method\\Hot Keys\\%08lX", dwID);
+        error = RegDeleteKeyW(HKEY_CURRENT_USER, szName);
+        return (error == ERROR_SUCCESS);
+    }
+
+    // Open "Control Panel"
+    error = RegCreateKeyExW(HKEY_CURRENT_USER, L"Control Panel", 0, NULL, 0, KEY_ALL_ACCESS,
+                            NULL, &hControlPanel, NULL);
+    if (error != ERROR_SUCCESS)
+        return FALSE;
+
+    // Open "Input Method"
+    error = RegCreateKeyExW(hControlPanel, L"Input Method", 0, NULL, 0, KEY_ALL_ACCESS,
+                            NULL, &hInputMethod, NULL);
+    if (error != ERROR_SUCCESS)
+        goto Quit;
```

Comment:
nit: I would prefer a nested control flow here, instead of goto.
In my opinion this communicates the control flow better.
```
if (RegCreateKey(hKey) == ERROR_SUCCESS)
{
    if (RegCreateKey(hkey2) == ERROR_SUCCESS)
    {
        // more
        RegCloseKey(hKey2);
    }
    RegCloseKey(hKey);
}
```

Reply:
OK.